pkg/test/executor/test_runner.go

```diff
@@ -115,7 +103,12 @@ func RunTestCase(
 
 	_, err = executor.Run(ctx, runCommandArguments)
 	if testCase.SubmitChecker != nil {
-		err = testCase.SubmitChecker(&job, err)
+		// TODO not sure how this behavior should be replicated.
+		err = testCase.SubmitChecker(&apimodels.PutJobResponse{
+			JobID:        job.ID,
+			EvaluationID: "TODO",
+			Warnings:     nil,
+		}, err)
 		require.NoError(t, err)
 	}
```

Comment:
`SubmitChecker` accepts a job and an error, and validate the result (i.e. err) of the submission is as expected. It can work with actually submitting jobs to an endpoint, or calling the executor directly as it only validates the error

Reply:
I see what you did. Makes sense. Lets keep it